### DISCUSSION TITLE
Get more logs on "Failed to decode"

### DISCUSSION THREAD

**Discussion by Piocky** [MusicPlayerDaemon/MPD#2075] [Q&A] [answered]
Hi,

Does anyone know how to get more logs from MPD? I get "Failed to decode" error when I try to play from an URL (mpc add then mpc play), but I can't get any specific information about the problem from the logs (either journalctl or /var/log/mpd.log with --verbose option)
I know it's not related to the URL itself, as it's working well on my Raspbian dev card with MPD 0.22.6, I just can't play any URL on my Yocto card, I always get this failed to decode error. I guess I miss one package or dependency, or I've got a difference in the config, but without more info it's really hard to find the exact cause..

The output of mpd --version on my Yocto image:
```
Music Player Daemon 0.23.15 (v0.23.15)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2021 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Database plugins:
 simple

Storage plugins:
 local udisks curl

Neighbor plugins:
 udisks

Decoders plugins:
 [mpg123] mp3
 [vorbis] ogg oga
 [oggflac] ogg oga
 [flac] flac
 [opus] opus ogg oga
 [dsdiff] dff
 [dsf] dsf
 [hybrid_dsd] m4a
 [wavpack] wv
 [pcm]

Filters:


Tag plugins:
 id3tag

Output plugins:
 null fifo pipe alsa pulse snapcast

Input plugins:
 file alsa curl

Playlist plugins:
 extm3u m3u pls xspf asx rss soundcloud flac cue embcue

Protocols:
 file:// alsa:// ftp:// ftps:// http:// https://

Other features:
 avahi dbus udisks epoll icu inotify ipv6 systemd tcp un
```

The [mpd.conf](https://github.com/user-attachments/files/16158542/mpd.conf.txt)

The relevant (but missing) logs from /var/log/mpd.log:
```
Jul 10 10:28 : client: [26] opened from local
Jul 10 10:28 : client: [26] process command list
Jul 10 10:28 : client: process command "add "https://listen.radioking.com/radio/651412/stream/714916""
Jul 10 10:28 : client: command returned 0
Jul 10 10:28 : client: [26] process command list returned 0
Jul 10 10:28 : client: [26] closed
Jul 10 10:28 : client: [27] opened from local
Jul 10 10:28 : client: [27] process command "play"
Jul 10 10:28 : playlist: play 0:"https://listen.radioking.com/radio/651412/stream/714916"
Jul 10 10:28 : client: [27] command returned 0
Jul 10 10:28 : client: [27] process command list
Jul 10 10:28 : client: process command "status"
Jul 10 10:28 : client: command returned 0
Jul 10 10:28 : client: process command "currentsong"
Jul 10 10:28 : client: command returned 0
Jul 10 10:28 : client: [27] process command list returned 0
Jul 10 10:28 : client: [27] closed
Jul 10 10:28 : curl: icy-metaint=16000
Jul 10 10:28 : exception: Failed to decode https://listen.radioking.com/radio/651412/stream/714916
Jul 10 10:28 : player: played "https://listen.radioking.com/radio/651412/stream/714916"
Jul 10 10:28 : playlist: stop
Jul 10 10:30 : state_file: Saving state file /data/mpd/state
```

**Comment by MaxKellermann** [comment]
In any case, you shouldn't be using MPD 0.22.6. It is three and a half years old.

**Reply by Piocky** [reply]
Yes I know, but what I meant is that it does work on the 0.22.6 on Raspbian, but not on the latest 0.23.15 on my Yocto image.

**Reply by Piocky** [reply]
@MaxKellermann can you help me on the initial question about the logs? It's not related to the MPD version, as I'm having the issue on the last version, and I don't think it's a problem related to MPD itself (that's why I opened a discussion and not an issue), but on the config or installation of the package, which is way more complex on Yocto than on Raspbian.

I first used the recipe from [here](https://git.openembedded.org/meta-openembedded/tree/meta-multimedia/recipes-multimedia/musicpd/mpd_0.23.12.bb?h=kirkstone), and then changed the SRCREV to get the 0.23.15 vers instead of 0.23.12, but I feel like the recipe misses some more stuff.

**Reply by MaxKellermann** [reply]
> as I'm having the issue on the last version

Then you should write a bug report. So far, you did not say that this affects the latest version, only that ancient 0.22 version, and I won't even look at reports for older versions.

**Reply by Piocky** [reply]
I only said that it's working well on the 0.22, that's it. Just because I mentioned it it means that you can't help me anyhow? I gave all the required information from your help page, and as you can see, I do use the latest MPD version, which is the 0.23.15.

**Reply by Piocky** [reply]
Again, I didn't open a bug report because I don't see this as an issue related to MPD, but if you prefer me to open one I would no problem.

**Reply by MaxKellermann** [reply]
I think we misunderstand. 0.22.6 was the only version in your post. I did not say I cannot help you with 0.23.15; I only said I cannot help with 0.22. And I told you to write a bug report.

**Reply by MaxKellermann** [reply]
Your build of MPD cannot play the stream because you have not enabled a decoder plugin which would be able to decode it. That part is indeed not a MPD bug, but a difference between what you want MPD to do and what you allowed MPD to do.
But what is a MPD bug is that MPD doesn't log anything helpful about this.

**Reply by Piocky** [reply]
Ok, can you tell me which one(s) is/are missing exactly? I do have several decoder plugins right?
Yes indeed, it would be very nice to have more info in this particular case, I looked everywhere I could, but couldn't find anything to help.

**Reply by MaxKellermann** [reply, marked as answer]
You have no plugin which can decode mp3 streams. You have the `mpg123` plugin but it can only decode local files. Candidates would be `mad` and `ffmpeg`.

**Reply by MaxKellermann** [reply]
Note: in MPD 0.24 (git master), `mpg123` supports streaming.

**Reply by Piocky** [reply]
Ok thank you very much! I will try to add those to my Yocto image, I'll let you know how it goes.

**Reply by Piocky** [reply]
I added ffmpeg, and it's working now! Thanks a lot for your help and reactivity 🙏
